src/ansys/tools/versioning/utils.py

```diff
@@ -276,3 +281,122 @@ def wrapper(self, *args, **kwargs):
         return wrapper
 
     return decorator
+
+
+class Version(int):
+    """Class for version comparison.
+
+    This class can be instantiated from a string or an integer.
+    The constructor will choose the corresponding class.
+
+    Examples
+    --------
+    >>> from ansys.tools.versioning.utils import Version
+    >>> Version(1)
+    1
+    >>> Version("dev")
+    'dev'
+    >>> Version(1) <= Version("dev")
+    True
+    >>> Version(99999) >= Version("dev")
+    False
+    """
+
+    def __new__(cls, value):
+        """Create and return a new object.
+
+        Args:
+            value (str or int):
+
+        Returns:
+            str, int: Returns a subclass of str or int depending on value.
+        """
+        if isinstance(value, str):
+            if value.strip().isdigit():
+                return super().__new__(cls, int(value.strip()))
+            else:
+                return str.__new__(str, value)
+        elif isinstance(value, int):
+            return super().__new__(cls, value)
+
+    def __le__(self, __x: int) -> bool:
+        """Less equal.
+
+        If compared against a string which contains 'dev' it will always evaluate to True.
+        If compared against an int, it will perform a classic 'less equal' operation.
+        """
+        if isinstance(__x, str):
+            if "dev" in __x:
+                return True
+            else:
+                raise ValueError("Invalid version string")
+        else:
+            return super().__le__(__x)
+
+    def __lt__(self, __x: int) -> bool:
```

Comment:
If I'm not mistaken the ``__x`` parameter can also be a str right? This will have to be adapted in the rest of the comparators.

Reply:
On it!

Reply:
I think it is fixed now.